src/image_utils.py

```diff
@@ -81,3 +82,33 @@ def varonRatio(S, B, c):
     std_deviation = np.std(ratio)
 
     return mean, std_deviation
+
+"""
+Generates a bounding box around each blob present within a binary label mask.
+The label mask may contain 0-n blobs, where n is the number of blobs present in the mask.
+
+Args:
+    label_mask (np.ndarray): 2D numpy array of 0s and 1s
+
+Returns:
+    np.ndarray: numpy array of bounding boxes, where each bounding box is represented 
+    as a tuple of (top-left, top-right, bottom-left, bottom-right), and each of these 
+    points is represented as a tuple of (x, y) coordinates.
+"""
+
+def binary_bbox(label_mask):
+    assert label_mask.ndim == 2, "labelMask must be a 2D numpy array"
+    
+    labelled_arr = measure.label(label_image=label_mask, connectivity=2)
+    bboxes = []
+    
+    for region in measure.regionprops(labelled_arr):
+        min_row, min_col, max_row, max_col = region.bbox
+        top_left = (min_col, min_row)
+        top_right = (max_col - 1, min_row)
+        bottom_left = (min_col, max_row - 1)
+        bottom_right = (max_col - 1, max_row - 1)
+        
+        bboxes.append((top_left, top_right, bottom_left, bottom_right))
```

Comment:
Since we assume straight rectangular boxes, top_left can be represented by just the min_col, top_right by max_col, and so on. That should reduce output parameters for the model

Reply:
While I do see what you're trying to do, I'm not sure how we'd be able to capture the box's height with this information, as the column just stores the x coordinate for the top right and top left corners, not the y coordinate. We won't know how tall the box is unless we have the y-coordinates for the top and bottom of the box. If we don't foresee the box corners being used every time the model runs/the bbox function is called, maybe it would be better to just return the min_row, min_col, max_row, max_col fields? Please let me know your thoughts.

Reply:
I think the height can be captured with the row info using y-top and y-bottom
Consider the following case:
```
[ 
  [0, 0, 0, 0, 0],
  [0, 1, 1, 1, 0],
  [0, 1, 1, 0, 0],
  [0, 1, 0, 0, 0],
  [0, 1, 0, 0, 0],
]
```

The box can be defined by x-left: 1, x-right: 3, y-top: 1, y-bottom: 4

Currently, it seems like the function returns top-left [1, 1], top-right [1, 3], bottom-left [4, 1], bottom-right [4, 3]

Reply:
Makes sense, thanks! I'll modify it to return tuples of the x-left, x-right, y-top, and y-bottom coordinates.